src/game.cpp

```diff
@@ -4050,15 +4052,34 @@ Creature *game::is_hostile_nearby()
     return is_hostile_within( distance );
 }
 
-Creature *game::is_hostile_very_close()
+Creature *game::is_hostile_very_close( bool dangerous )
 {
-    return is_hostile_within( DANGEROUS_PROXIMITY );
+    return is_hostile_within( DANGEROUS_PROXIMITY, dangerous );
 }
 
-Creature *game::is_hostile_within( int distance )
+Creature *game::is_hostile_within( int distance, bool dangerous )
 {
     for( auto &critter : u.get_visible_creatures( distance ) ) {
         if( u.attitude_to( *critter ) == Creature::Attitude::HOSTILE ) {
+            if( dangerous ) {
+                if( critter->has_flag( MF_RANGED_ATTACKER ) ) {
+                    return critter;
+                }
+
+                for( const std::pair<std::string, mtype_special_attack> &attack :
+                     critter->as_monster()->type->special_attacks ) {
+                    if( attack.second->id == "gun" ) {
+                        return critter;
+                    }
+                }
+
+                const pathfinding_settings pf_settings = pathfinding_settings { 8, DANGEROUS_PROXIMITY, DANGEROUS_PROXIMITY * 2, 4, true, false, true, false, false };
```

Comment:
Use the `distance` argument here instead of `DANGEROUS_PROXIMITY`.

Reply:
Ok, done in https://github.com/CleverRaven/Cataclysm-DDA/pull/45752/commits/6eae1c4f8f6287b4c87dd980784f0ddd051614c8.